### ISSUE TITLE
Feature request: Transparent window or even transparent + shaped

### ISSUE THREAD

**Issue by KeyWeeUsr** [kivy/kivy#4307]
A long ago I saw [LiLi USB creator](http://www.linuxliveusb.com) and its Window kind of surprised me. Not only it's transparent, it's based on a bitmap(or at least another type of image) too. Since then it kind of bugs me how to make it - now with Kivy, but still no success. However I found some scraps, that may help someone who would like to code it.

On irc @dessant pointed me to these links [wiki.libsdl.org](https://wiki.libsdl.org/SDL-gsoc2010_shaped_windows?highlight=%28windowshape%29), [SDL-mirror](https://github.com/spurious/SDL-mirror/blob/e1e235cb9012f8edc705d2cb61df9e8a5b0a4047/include/SDL_shape.h#L120)

LiLi provides a [portable](http://www.linuxliveusb.com/downloads/?version=stable-portable) package, where is its source available. In `Graphics.au3` you can see that the guy uses Win32API, which probably isn't desired when there is sdl2 and still a little bit of pygame present with Android, but it may give you a picture what it is exactly. This type of Window might drag new users to Kivy if it'd be at least on these commercial platforms such as Windows and Mac.

_Links:_
- [C++ Semi-Transparent Window SDL](http://stackoverflow.com/questions/6809764/c-semi-transparent-window-sdl)
- [Transparent SDL2 Game](http://stackoverflow.com/questions/25038347/transparent-sdl2-game-win32-colorkeying-not-working-correctly)
- [Background from OS in Window](http://gamedev.stackexchange.com/questions/117792/why-does-sdl-render-the-background-from-the-operating-system-to-the-window)
- [Transparent window background](https://www.libgosu.org/cgi-bin/mwf/topic_show.pl?tid=1171)
- [SDL transparent Window](http://www.gamedev.net/topic/637618-sdl-transparent-window/) - this one kind of tells me that it may not be possible with sdl2
- [SDL transparent primary surface (main window)](http://forums.libsdl.org/viewtopic.php?p=44872&sid=b15423b1faaa0238e5fb0e6e6628de92#44872) - however this seems to be promising

**Comment by Fuzzyma** [comment]
I am also waiting for this feature. You can ofc mix `win32api` with `kivy` to make your windows transparent but thats ugly and only works for windows. It would be cool, to have some kivy syntax to actually design the window and not only the root widget.

Maybe:

```
<kivy:Window>:
    canvas:
        Color: 
            rgba: 1, 1, 1, 0
```

or just 

```
<kivy:Window>:
    borderless: True
    resizable: False
    window_state: visible
    background:
        rgba: 0, 0, 0, 1
    opacity: 0.5
```
In this case background and opacity would even be different things. Background would be the (at the moment) black screen you get, when you only open a window without anything. Could be also transparent with rgba.  
Opacity would make everyhing (including border and title) transparent

**Comment by BillBridge** [comment]
I am also looking forward to these features. Currently I am using a combination of win32api and kivy. Agree with suggestions of Fuzzyma.

**Comment by KeyWeeUsr** [comment]
@BillBridge If you could, you might want to submit a pull request for that behavior. Although it probably won't be merged due to working only with a single platform, it will be useful at least for Windows users and who knows, maybe someone makes an alternative for linux/mac.

I also noticed that on e.g. Ubuntu when there's a problem with not loaded window, there's a _transparent_ background on such window. If this behavior was tracked to what's actually hapenning, it could be used as a part of this feature. Or maybe there's just no background for a window on a linux by default... which again is an advantage for this feature.

**Comment by Fuzzyma** [comment]
To make a window transparent in windows (the whole - not the background) you need to use the win32 api. The code for this would be as follows:

```python
import win32gui

# Get the window
handle = win32gui.FindWindow(None, "TitleOfYourWindow")

# Make it a layered window
win32gui.SetWindowLong(handle, win32con.GWL_EXSTYLE, win32gui.GetWindowLong(self.handle, win32con.GWL_EXSTYLE) | win32con.WS_EX_LAYERED)

# make it transparent (alpha between 0 and 255)
win32gui.SetLayeredWindowAttributes(handle, win32api.RGB(0, 0, 0), alpha, win32con.LWA_ALPHA)
```

That said, doing a PR is no fun if 
> it probably won't be merged

However: I am also not capable of doing it because I am not familiar with the kivy code

**Comment by KeyWeeUsr** [comment]
Few more links I found recently:

* Mac: [sample](https://developer.apple.com/library/content/samplecode/RoundTransparentWindow/Introduction/Intro.html)
* Linux [Glass Pane](http://stackoverflow.com/q/4326534/5994041)
* [OpenGL](http://stackoverflow.com/q/4052940/5994041)

With Windows and Mac it looks promising, but most likely requires a lot of time.

**Comment by KeyWeeUsr** [comment]
[Android](http://stackoverflow.com/a/2700683/5994041) seems quite possible too, except for the shaped window. However, I'm not sure if it'd have any use-case on android anyway.

**Comment by botzill** [comment]
Hey Guys!

I want to ask if there are any progress related to this feature? I'm also interested into it.

Thx!

**Comment by KeyWeeUsr** [comment]
Closed by #5252
